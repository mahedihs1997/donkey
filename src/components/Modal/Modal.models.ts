import { ModalProps as MuiModalProps } from '@material-ui/core';

export interface ModalProps extends MuiModalProps {
  open: boolean;
  onClose: () => void;
}
