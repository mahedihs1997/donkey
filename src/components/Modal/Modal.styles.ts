import { makeStyles } from '@material-ui/core/styles';
import theme from '@theme/mui';

export const useStyles = makeStyles({
  paper: {
    position: 'absolute',
    width: 400,
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(2, 4, 3),
  },
});

export const modalStyle = {
  top: '50%',
  left: '50%',
  transform: `translate(-50%, -50%)`,
};
