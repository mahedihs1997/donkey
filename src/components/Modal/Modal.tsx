import React from 'react';
import { Modal as MuiModal } from '@material-ui/core';
import { ModalProps } from './Modal.models';
import { modalStyle, useStyles } from './Modal.styles';

const Modal: React.FunctionComponent<ModalProps> = (props) => {
  const { children, handleClose } = props;
  const classes = useStyles();

  return (
    <MuiModal onClose={handleClose} {...props}>
      <div style={modalStyle} className={classes.paper}>
        {children}
      </div>
    </MuiModal>
  );
};

export default Modal;
