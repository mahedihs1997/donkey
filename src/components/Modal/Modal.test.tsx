import React from 'react';
import TestRenderer from 'react-test-renderer';
import Modal from '.';

describe('<Modal />', () => {
  it('should render without throwing an error and match snapshot', () => {
    const testRenderer = TestRenderer.create(
      <Modal open={true} handleClose={jest.fn()}>
        <div>Ugendo</div>
      </Modal>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
