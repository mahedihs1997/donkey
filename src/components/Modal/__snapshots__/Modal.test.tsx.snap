// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render without throwing an error and match snapshot 1`] = `null`;
